### DISCUSSION TITLE
When modeling with @observable decorator

### DISCUSSION THREAD

**Discussion by codemilli** [mobxjs/mobx#3288] [Q&A] [answered]
Hello all, 
I am building a Canvas Image Editor with MobX, and I use decorator to make state observable. 
Whenever changes detected, autorun is called and read every properties from model and draw canvas application. 

I want autorun is called when image is loaded, so I did modeling like below

```ts
class A {
  @observable
  private _image: HTMLImageElement;

  public getImage() {
    return this._image;
  }
}
```

For me, making HTMLImageElement type observable data seems not that good, is this valid or is there any better modeling?

**Comment by kubk** [comment, marked as answer]
Mobx skips converting unknown classes. You can check it by running `console.log(isObservable(this._image))`. It'll give `false`.

> I want autorun is called when image is loaded, so I did modeling like below

No need to use autorun for that. You can collocate event with reaction like this:

```
loadImage(image: HTMLImageElement) {
  // run code from autorun
}
```

**Reply by codemilli** [reply]
"Mobx skips converting unknown classes." now I got it. Thank you. 
So mobx observes only the reference change?

**Reply by codemilli** [reply]
In my case, I have to place that logic in the autorun.

**Reply by kubk** [reply]
> So mobx observes only the reference change?

@codemilli Mobx just ignores unknown classes, so they don't become observable at all.

**Reply by codemilli** [reply]
So, if I wanna observe whether image is loaded and get loaded image in `autorun`, I need another flags like below, do I?
```ts
class A {
  private _image: HTMLImageElement;
  
  @observable
  private _imageLoaded = false;
  
  @computed
  public get image() {
    return this._imageLoaded ? this._image : null;
  };  

  public loadImage() { ....  }
}
```

**Reply by kubk** [reply]
@codemilli That should work 👍
